### PR TITLE
Reduce the heartbeat size of VerifyReplicationTasks activity

### PR DESCRIPTION
**Pull request by hehaifengcn**
**What changed?**

Reducing heart beat size (current payload size is 26K for 1000 executions). 

**Why?**

**How did you test it?**
unit tests + cluster tests 

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

File: service/worker/migration/activities.go [hunk 1]
```diff
@@ -567,44 +538,67 @@ func (a *activities) verifyHandleNotFoundWorkflow(
...
+			a.forceReplicationMetricsHandler.Counter(metrics.EncounterNotFoundWorkflowCount.GetMetricName()).Record(1)
+			a.logger.Info("createReplicationTasks skip NotFound workflow", tags...)
+			return true, reasonWorkflowNotFound, nil
 		}
 
-		return err
+		return false, "", err
 	}
 
+	// Zombie workflow should be a transient state. However, if there is Zombie workflow on the source cluster,
```
Comment (meiliang86) [inline service/worker/migration/activities.go:550]: We might need to revisit this when we do GN and migration. This is not true any more for GN.

File: service/worker/migration/activities.go [hunk 2]
```diff
@@ -553,12 +525,11 @@ func isNotFoundServiceError(err error) bool {
...
 
-func (a *activities) verifyHandleNotFoundWorkflow(
+func (a *activities) canSkipWorkflowExecution(
 	ctx context.Context,
 	namespaceID string,
 	we *commonpb.WorkflowExecution,
-	result *VerifyResult,
-) error {
+) (bool, string, error) {
 	tags := []tag.Tag{tag.WorkflowType(forceReplicationWorkflowName), tag.WorkflowNamespaceID(namespaceID), tag.WorkflowID(we.WorkflowId), tag.WorkflowRunID(we.RunId)}
```
Comment (meiliang86) [inline service/worker/migration/activities.go:533]: nit: forceReplicationWorkflowName does not seem to be a useful tag

File: service/worker/migration/activities.go [hunk 3]
```diff
@@ -675,34 +661,53 @@ func (a *activities) VerifyReplicationTasks(ctx context.Context, request *verify
...
-	//  - more than RetryableTimeout, the activity fails with retryable error and activity will restart. This gives us the chance
-	//    to identify case #2 and #3 by rerunning createReplicationTasks.
-	//  - more than NonRetryableTimeout, it means potentially we encountered #4. The activity returns
-	//    non-retryable error and force-replication workflow will restarted.
-	for {
+	//  - more than checkSkipThreshold, it checks if outstanding workflow execution can be skipped locally (#2 and #3)
+	//  - more than NonRetryableTimeout, it means potentially #4. The activity returns
+	//    non-retryable error and force-replication will fail.
 
+	checkSkipCount := 0
```
Comment (meiliang86) [inline service/worker/migration/activities.go:668]: nit: the name of this variable is a little bit hard to follow
If you cannot think of a better name, at least leave a comment explaining why you want to do this.

**Review by meiliang86 — APPROVED** [review]

File: service/worker/migration/activities.go (hunk 3 above)
Comment (hehaifengcn) [reply service/worker/migration/activities.go]: code removed.

**Review by meiliang86 — APPROVED** [review]
